src/main/java/com/snowplowanalytics/redash/RedashClient.java

```diff
@@ -0,0 +1,422 @@
+     */
+    public List<DataSource> getDataSources() throws IOException {
+        String url = baseUrl + DATA_SOURCES + APIKEY_STRING + apiKey;
+        Type listType = new TypeToken<ArrayList<DataSource>>() {}.getType();
+        return new Gson().fromJson(get(url), listType);
+    }
+
+    /**
+     * Removes data-source
+     *
+     * @param dataSourceId id of data-source which will be deleted if such presents.
+     * @return false if data-source with provided id doesn't exist on the server, if it does then data-source will be
+     * deleted and true will be returned.
+     * @throws IOException if server is not available due to connection error or API key is invalid.
+     */
+    public boolean deleteDataSource(int dataSourceId) throws IOException {
+        String url = baseUrl + DATA_SOURCES + "/" + dataSourceId + APIKEY_STRING + apiKey;
+        return delete(url).isEmpty();
+    }
+
+    /**
+     * Creates user-group onto Redash server
+     *
+     * @param userGroup should contain correct value name field, as it will be used in creation.
+     * @return id of newly created group, if there wasn't with such name, and in this case argument's id field will
+     * be set by value that was generated and returned by server in the response body.
+     * @throws IllegalArgumentException if there is already exists user-group with the name provided inside dataSource
+     * @throws IOException              if server is not available due to connection error or API key is invalid.
+     */
+    public int createUserGroup(UserGroup userGroup) throws IOException {
+        if (isEntityAlreadyExists(getUserGroups(), userGroup.getName())) {
+            throw new IllegalArgumentException(USER_GROUP_ALREADY_EXISTS);
+        }
+        String url = baseUrl + GROUPS + APIKEY_STRING + apiKey;
+        String returnValue = post(url, new Gson().toJson(userGroup));
+        int id = getIdFromJson(returnValue);
+        userGroup.setId(id);
+        return id;
+    }
+
+    /**
+     * Adds user into user-group both specified by their ids.
+     *
+     * @return false if specified user-group already contains specified user, true if not and user was successfully added.
+     * @throws IllegalArgumentException if no user or no user group with provided userId and userGroupId accordingly exist
+     *                                  onto Redash server.
+     * @throws IOException              if server is not available due to connection error or API key is invalid.
+     */
+    public boolean addUserToGroup(int userId, int groupId) throws IOException {
+        User user = getUserById(userId);
+        checkIfUserGroupExists(groupId);
+        if (user.getGroups().contains(groupId)) {
+            return false;
+        }
+        String url = baseUrl + GROUPS + "/" + groupId + MEMBERS + APIKEY_STRING + apiKey;
+        String json = "{\"user_id\":" + userId + "}";
```

Comment:
I'd avoid working with JSON strings directly. Create a `JSONObject`, add fields to it as needed, _then_ convert it to a JSON string.